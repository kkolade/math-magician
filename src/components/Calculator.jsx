import PropTypes from 'prop-types';
import React from 'react';

function Calculator({ initialValue }) {
  return (
    <div className="calc-container">
      <input className="display" type="text" value={initialValue} />
      <button type="submit" className="calc-key">
        A/C
      </button>
      <button type="submit" className="calc-key">
        +/-
      </button>
      <button type="submit" className="calc-key">
        %
      </button>
      <button type="submit" className="operator">
        /
      </button>
      <button type="submit" className="calc-key">
        7
      </button>
      <button type="submit" className="calc-key">
        8
      </button>
      <button type="submit" className="calc-key">
        9
      </button>
      <button type="submit" className="operator">
        *
      </button>
      <button type="submit" className="calc-key">
        4
      </button>
      <button type="submit" className="calc-key">
        5
      </button>
      <button type="submit" className="calc-key">
        6
      </button>
      <button type="submit" className="operator">
        -
      </button>
      <button type="submit" className="calc-key">
        1
      </button>
      <button type="submit" className="calc-key">
        2
      </button>
      <button type="submit" className="calc-key">
        3
      </button>
      <button type="submit" className="operator">
        +
      </button>
      <button type="submit" className="zero">
        0
      </button>
      <button type="submit" className="calc-key">
        .
      </button>
      <button type="submit" className="operator">
        =
      </button>
    </div>
  );
}

Calculator.propTypes = { initialValue: PropTypes.number.isRequired };

export default Calculator;
