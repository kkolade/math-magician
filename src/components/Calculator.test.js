import { cleanup, fireEvent, render } from '@testing-library/react';
import React from 'react';
import Calculator from './Calculator';

describe('Calculator', () => {
  afterEach(cleanup);

  test('renders correctly', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});

test('Calculator handles user input correctly', () => {
  const { getByText, queryByTestId } = render(<Calculator />);

  // Click on "1", "2", "3", "+", "4", "5", "6", "=", expect "579"
  fireEvent.click(getByText('1'));
  fireEvent.click(getByText('2'));
  fireEvent.click(getByText('3'));
  fireEvent.click(getByText('+'));
  fireEvent.click(getByText('4'));
  fireEvent.click(getByText('5'));
  fireEvent.click(getByText('6'));
  fireEvent.click(getByText('='));
  expect(queryByTestId('display')).toHaveTextContent('579');

  // Click on "AC", expect "0"
  fireEvent.click(getByText('AC'));
  expect(queryByTestId('display')).toHaveTextContent('0');

  // Click on "9", "-", "3", "=", expect "6"
  fireEvent.click(getByText('9'));
  fireEvent.click(getByText('-'));
  fireEvent.click(getByText('3'));
  fireEvent.click(getByText('='));
  expect(queryByTestId('display')).toHaveTextContent('6');

  // Click on "5", "x", "4", "/", "2", "+", "1", expect "11"
  fireEvent.click(getByText('5'));
  fireEvent.click(getByText('x'));
  fireEvent.click(getByText('4'));
  fireEvent.click(getByText('/'));
  fireEvent.click(getByText('2'));
  fireEvent.click(getByText('+'));
  fireEvent.click(getByText('1'));
  fireEvent.click(getByText('='));
  expect(queryByTestId('display')).toHaveTextContent('11');
});

afterEach(cleanup);

// import { fireEvent, render } from '@testing-library/react';
// import React from 'react';
// import renderer from 'react-test-renderer';
// import Calculator from './Calculator';

// test('Calculator component renders correctly', () => {
//   const component = renderer.create(<Calculator />);
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('adds 2 + 3 and displays 5', () => {
//   const { getByText, queryByTestId } = render(<Calculator />);
//   const display = queryByTestId('diplay');
//   const button2 = getByText('2');
//   const buttonPlus = getByText('+');
//   const button3 = getByText('3');
//   const buttonEquals = getByText('=');

//   fireEvent.click(button2);
//   fireEvent.click(buttonPlus);
//   fireEvent.click(button3);
//   fireEvent.click(buttonEquals);

//   expect(display.textContent).toBe('5');
// });
