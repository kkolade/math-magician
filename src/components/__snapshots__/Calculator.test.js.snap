// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<div
  className="calc-container"
>
  <div
    className="display"
    data-testid="display"
  >
    0
  </div>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    AC
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    +/-
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    %
  </button>
  <button
    className="operator"
    onClick={[Function]}
    type="submit"
  >
    รท
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    7
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    8
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    9
  </button>
  <button
    className="operator"
    onClick={[Function]}
    type="submit"
  >
    x
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    4
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    5
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    6
  </button>
  <button
    className="operator"
    onClick={[Function]}
    type="submit"
  >
    -
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    1
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    2
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    3
  </button>
  <button
    className="operator"
    onClick={[Function]}
    type="submit"
  >
    +
  </button>
  <button
    className="zero"
    onClick={[Function]}
    type="submit"
  >
    0
  </button>
  <button
    className="calc-key"
    onClick={[Function]}
    type="submit"
  >
    .
  </button>
  <button
    className="operator"
    onClick={[Function]}
    type="submit"
  >
    =
  </button>
</div>
`;

exports[`Calculator renders correctly 1`] = `
<div>
  <div
    class="calc-container"
  >
    <div
      class="display"
      data-testid="display"
    >
      0
    </div>
    <button
      class="calc-key"
      type="submit"
    >
      AC
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      +/-
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      %
    </button>
    <button
      class="operator"
      type="submit"
    >
      รท
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      7
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      8
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      9
    </button>
    <button
      class="operator"
      type="submit"
    >
      x
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      4
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      5
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      6
    </button>
    <button
      class="operator"
      type="submit"
    >
      -
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      1
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      2
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      3
    </button>
    <button
      class="operator"
      type="submit"
    >
      +
    </button>
    <button
      class="zero"
      type="submit"
    >
      0
    </button>
    <button
      class="calc-key"
      type="submit"
    >
      .
    </button>
    <button
      class="operator"
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;
